lib/std/hash/cityhash.zig

```diff
@@ -6,16 +6,22 @@
 const std = @import("std");
 const builtin = @import("builtin");
 
+fn offsetPtr(ptr: [*]const u8, offset: usize) [*]const u8 {
+    // ptr + offset doesn't work at comptime so we need this instead.
+    return @ptrCast([*]const u8, &ptr[offset]);
+}
+
 pub const CityHash32 = struct {
     const Self = @This();
 
     // Magic numbers for 32-bit hashing.  Copied from Murmur3.
     const c1: u32 = 0xcc9e2d51;
     const c2: u32 = 0x1b873593;
 
-    fn fetch32(ptr: [*]const u8) u32 {
-        var v: u32 = undefined;
-        @memcpy(@ptrCast([*]u8, &v), ptr, 4);
+    fn fetch32(ptr: [*]const u8, offset: usize) u32 {
+        // ptr + offset doesn't work at comptime so we need this instead.
+        const offset_ptr = offsetPtr(ptr, offset)[0..4];
```

Comment:
What about 
```zig
return mem.readIntNative(u32, offsetPtr(ptr, offset)[0..4]);
```

Ditto for `fetch64` (and the other two identical fetch helpers? Can you just pull them at top level and delete a copy?)

Reply:
Huh, I thought these wouldn't work at comptime but apparently they do.  Since the next line is byte swap, I can replace the whole function with `readIntLittle`.

`[offset .. offset + 4]` works at comptime but at runtime this would produce a slice, so I'm sticking with the other one.

Reply:
This should work at runtime:

```zig
return mem.readIntLittle(u32, ptr[offset .. offset + 4][0..4]);
```

(With slices, you can do `[offset..][0..4]` which is a nice little idiom, but that doesn't work for length-less pointers)

Reply:
That works but introduces a new but useless runtime check in debug modes (useless because it's only checking that `offset + 4 - offset == 4`).
`ptr[offset .. offset + 4].ptr[0..4]` would work, but for consistency I think it's worth sticking with `offsetPtr` everywhere in this file.